pandas/core/frame.py

```diff
@@ -4343,6 +4345,46 @@ def join(self, other, on=None, how='left', lsuffix='', rsuffix='',
         on, lsuffix, and rsuffix options are not supported when passing a list
         of DataFrame objects
 
+        Examples
+        --------
+        >>> caller = pd.DataFrame({'key': ['K0', 'K1', 'K2', 'K3', 'K4', 'K5'],
+        ...                        'A': ['A0', 'A1', 'A2', 'A3', 'A4', 'A5']})
+
+        >>> caller
+            A key
+        0  A0  K0
+        1  A1  K1
+        2  A2  K2
+        3  A3  K3
+        4  A4  K4
+        5  A5  K5
+
+        >>> other = pd.DataFrame({'key': ['K0', 'K1', 'K2'],
+        ...                       'B': ['B0', 'B1', 'B2']})
+
+        >>> other
+            B key
+        0  B0  K0
+        1  B1  K1
+        2  B2  K2
+
+        Perform a left join using caller's key column and other frame's index 
+
+        >>> caller.join(other.set_index('key'), on='key', how='left',
```

Comment:
What about having just these two examples:

Perform a left join using caller's key column and other frame's index 

`caller.join(other.set_index('key'), on='key', how='left')`

Set `key` as the index column on `caller` and `other` and perform an index-on-index join.

`caller.set_index('key').join(other.set_index('key'), how='left')`